id/src/logics/verificationLogic.ts

```diff
@@ -89,40 +121,63 @@ export const verificationLogic = kea<verificationLogicType>([
   listeners(({ actions, values }) => ({
     // ANCHOR connection listeners
     initConnection: async () => {
-      if (values.connectorUri) {
-        return
-      }
+      debug('Starting connection process')
+      debug(values.walletconnectProjectId)
 
-      if (connector.connected) {
-        try {
-          await connector.killSession()
-        } catch (error) {
-          console.error('Error while killing session', error)
-        }
-      }
+      const currentClient = await Client.init({
+        projectId: values.walletconnectProjectId,
+        metadata: {
+          name: 'World ID',
+          description: 'Verify with World ID',
+          url: '#',
+          icons: ['https://worldcoin.org/icons/logo-small.svg'],
+        },
+      })
+
+      debug('currentClient: ', currentClient)
+      actions.setClient(currentClient)
 
       try {
-        await connector.createSession()
-        actions.setConnectorUri(connector.uri)
-        actions.setConnectionStartTime(performance.now())
-      } catch (error) {
-        console.error('Error while creating WalletConnect session', error)
-      }
+        debug('Connecting to WalletConnect')
 
-      connector.on('connect', async (error: unknown) => {
-        if (error) {
-          actions.setError(ErrorCodes.ConnectionFailed)
-          console.error(`Failed to establish connection to WLD app: ${error}`)
-        } else {
-          actions.handleConnectionEstablished()
-        }
-      })
+        const { uri, approval } = await currentClient.connect({
+          requiredNamespaces: {
+            eip155: {
+              methods: ['wld_worldIDVerification'],
+              chains: ['eip155:1'], // Chain ID used does not matter, since we only perform signatures
```

Comment:
```suggestion
              chains: ['eip155:1'], // Chain ID used does not matter, since we only perform custom JSON RPC messages (World ID verification)
```